### PR TITLE
feat(core): modify exception handler test cases

### PR DESCRIPTION
**Pull request by jochongs**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #13550 #13870

## What is the new behavior?

It appears that the current exception-handler test cases do not utilize the updated `combineStackTrace` function.

```typescript
it('when exception is instanceof RuntimeException', () => {
  const exception = new RuntimeException('msg');
  instance.handle(exception);
  expect(errorSpy.calledWith(exception.message, exception.stack)).to.be
    .true;
});
```

After reviewing the PR, I noticed that in cases where the exception is a `RuntimeException`, the `ExceptionHandler.logger.error` method uses `combineStackTrace(exception)` as the second parameter.

```typescript
public handle(exception: RuntimeException | Error) {
  if (!(exception instanceof RuntimeException)) {
    ExceptionHandler.logger.error(
      exception.message,
      combineStackTrace(exception),
    );
    return;
  }
  ExceptionHandler.logger.error(
    exception.what(),
    combineStackTrace(exception),
  );
}
```

To align with this change, I have made some modifications to the related test code and am submitting this PR. Please review the changes.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

**Comment by thibseisel** [comment]
Hi @jochongs, thanks for your contribution!
I intentionally did not update these tests, as they ensure that my change preserves the exact same behavior in case of an error without a cause. 
We could add a third test asserting that given an Error with a cause, `Logger.error` is called with `combineStackTrace(error)` as its second argument. What do you think?

**Comment by jochongs** [comment]
I understand that these test cases are designed to verify the `handle` method of the `ExceptionHandler`. Given this, I believe it might be unnecessary to specifically check whether `combineStackTrace` correctly handles the cause value within these particular tests.

As it stands, the `handle` method wraps the `exception` with `combineStackTrace` and passes it as the second argument, regardless of whether the `Error` has `cause`. I believe the behavior of `combineStackTrace` regarding the cause is sufficiently covered in the existing test case:

https://github.com/thibseisel/nest/blob/4d37a0b1a83e2da0c682b7842f66c25965827500/packages/core/test/helpers/combine-stack-trace.spec.ts#L13-L37

Additionally, the second test case I've modified is intended to demonstrate how the `handle` method behaves when the `exception` is not an instance of `RuntimeException`. However, I noticed that the object being passed as an `exception`, `InvalidMiddlewareException`, actually inherits from `RuntimeException`, which seems to contradict the intended purpose of testing non-RuntimeException cases.

In conclusion, since these test cases I have modified are focused on testing the `handle` method itself, I believe it is sufficient to have two test cases: one where the `exception` is a `RuntimeException` and one where it is not.
